tools/release/build-xcframeworks.sh

```diff
@@ -0,0 +1,151 @@
+
+    [ -f "Cartfile.resolved" ] && echo_succ "Found 'Cartfile.resolved' in '$REPO_PATH'" \
+        || { echo_err "Error:" "Could not find 'Cartfile.resolved' in '$REPO_PATH'." ; exit 1; }
+
+    local config_files=$(find . -name "*.local.xcconfig")
+    if [[ -n $config_files ]]; then
+        echo_err "Error:" "The repo at '$REPO_PATH' is not in a clean state."
+        echo "It has following local config files:"
+        echo "$config_files" | awk '{print "- " $0}'
+        exit 1
+    else
+        echo_succ "The repository is in a clean state and no '*.local.xcconfig' files are present."
+    fi
+}
+
+archive() {
+    local scheme="$1"
+    local destination="$2"
+    local archive_path="$3"
+
+    echo_subtitle2 "➔ Archive scheme: '$scheme' for destination: '$destination'"
+
+    xcodebuild archive \
+        -workspace "Datadog.xcworkspace" \
+        -scheme $scheme \
+        -destination $destination \
+        -archivePath $archive_path \
+        SKIP_INSTALL=NO \
+        BUILD_LIBRARY_FOR_DISTRIBUTION=YES \
+        ONLY_ACTIVE_ARCH=NO \
+        | xcbeautify
+
+    echo_succ "The archive was created successfully at '$archive_path.xcarchive'"
+}
+
+build_xcframework() {
+    local product="$1"
+    local platform="$2"
+    xcoptions=()
+
+    echo_subtitle2 "Build '$product.xcframework' using platform='$platform'"
+
+    if [[ $platform == *"iOS"* ]]; then
+        echo "▸ Archive $product iOS"
+
+        archive "$product iOS" "generic/platform=iOS" "$ARCHIVES_TEMP_OUTPUT/$product/ios"
+        xcoptions+=(-archive "$ARCHIVES_TEMP_OUTPUT/$product/ios.xcarchive" -framework "$product.framework")
+
+        archive "$product iOS" "generic/platform=iOS Simulator" "$ARCHIVES_TEMP_OUTPUT/$product/ios-simulator"
+        xcoptions+=(-archive "$ARCHIVES_TEMP_OUTPUT/$product/ios-simulator.xcarchive" -framework "$product.framework")
+    fi
+
+    if [[ $platform == *"tvOS"* ]]; then
+        echo "▸ Archive $product tvOS"
+
+        archive "$product tvOS" "generic/platform=tvOS" "$ARCHIVES_TEMP_OUTPUT/$product/tvos"
+        xcoptions+=(-archive "$ARCHIVES_TEMP_OUTPUT/$product/tvos.xcarchive" -framework "$product.framework")
+
+        archive "$product tvOS" "generic/platform=tvOS Simulator" "$ARCHIVES_TEMP_OUTPUT/$product/tvos-simulator"
+        xcoptions+=(-archive "$ARCHIVES_TEMP_OUTPUT/$product/tvos-simulator.xcarchive" -framework "$product.framework")
+    fi
+
+    # Datadog class conflicts with module name and Swift emits invalid module interface
+    # cf. https://github.com/apple/swift/issues/56573
+    #
+    # Therefore, we cannot provide ABI stability and we have to supply '-allow-internal-distribution'.
+    xcodebuild -create-xcframework -allow-internal-distribution ${xcoptions[@]} -output "$XCFRAMEWORKS_OUTPUT/$product.xcframework" | xcbeautify
+
+    echo_succ "The '$product.xcframework' was created successfully in '$XCFRAMEWORKS_OUTPUT'"
+}
+
+echo_info "cd '$REPO_PATH'"
+cd $REPO_PATH
+
+check_repo_clean_state
```

Comment:
in what case this can fail in CI?

Would it makes sense to have an option to clear the repo state which is `true` in CI and `false` for our local testing?

Reply:
✅ Replaced with `git clean -fxd` inside `artifacts/<GIT_TAG>/dd-sdk-ios`. It's still defensive (the repo is cloned so must be clean), but I'd like to keep it so we can leverage this if we decide to use current repo for building artifacts as discussed in other comments.